flocker/control/_protocol.py

```diff
@@ -734,6 +749,7 @@ def connectionMade(self):
         AMP.connectionMade(self)
         self.agent.connected(self)
         self._pinger.start(self, PING_INTERVAL)
+        # Send SetNodeEra using value of /proc/sys/kernel/random/boot_id
```

Comment:
If someone continues to deploy using our container agent, then is it the case that this design fixes the container agent misbehavior because in the previously problematic case where the container agent connected first and saw the dataset agent state from the previous boot, it would proceed to act incorrectly whereas now when it first connects it will send `SetNodeEra` with the new value which will wipe the dataset agent state from the previous boot, preventing it from seeing it and taking incorrect actions as a result?

I expected to see some era comparison in the `IDeployer.calculate_actions` implementation for the container agent too.  Perhaps those would be redundant with the fix here and that's why they're not in the design?  OTOH perhaps that redundancy wouldn't necessarily be bad.

Anyhow, have I understood correctly?